server/utils/hollaex-tools-lib/tools/user.js

```
@@ -2345,6 +2366,605 @@ const changeKitUserEmail = async (userId, newEmail, auditInfo) => {
+			const filteredFees = {};
+			const referralHistory = [];
+			if (count === 0) {
+				throw new Error('No trades made by affiliated users');
+			}
+
+			loggerUser.verbose(
+				'REFERRAL createUnrealizedReferralFees',
+				'Affiliated users that traded:',
+				count
+			);
+
+			for (let affiliation of affiliations) {
+				const refereeUser = affiliation.user;
+				const referer = affiliation.referer;
+
+				if (accumulatedFees[refereeUser.network_id]) {
+					// refererKey includes user kit id, user network id, and user email separated by colons
+					const refererKey = `${referer.id}:${referer.network_id}:${referer.email}`;
+					if (!filteredFees[refererKey]) {
+						filteredFees[refererKey] = {};
+					}
+
+					for (let coin in accumulatedFees[refereeUser.network_id]) {
+						if (!isNumber(filteredFees[refererKey][coin])) {
+							filteredFees[refererKey][coin] = 0;
+						}
+
+						filteredFees[refererKey][coin] = addAmounts(
+							filteredFees[refererKey][coin],
+							accumulatedFees[refereeUser.network_id][coin]
+						);
+
+
+						const refIndex = referralHistory.findIndex(data => data.referee === refereeUser.id && data.referer === referer.id && data.coin === coin);
+						if (refIndex >= 0) {
+							referralHistory[refIndex].accumulated_fees = filteredFees[refererKey][coin];
+						} else {
+							referralHistory.push({
+								referer: referer.id,
+								referee: refereeUser.id,
+								timestamp: lastSettledTrade,
+								code: affiliation.code,
+								earning_rate: affiliation.earning_rate,
+								coin,
+								accumulated_fees: filteredFees[refererKey][coin]
+							});
+						}
+					}
+				}
+			}
+
+			referralHistory.forEach(refHistory => {
+				refHistory.accumulated_fees = applyEarningRate(refHistory.accumulated_fees, refHistory.earning_rate);
+			});
+
+
+			return referralHistory;
+		})
+		.then(async (referralHistory) => {
+			const nativeCurrency = getKitConfig()?.referral_history_config?.currency || 'usdt';
```

if this currency is not defined we should throw an error.